src/vcpkg/postbuildlint.cpp

```diff
@@ -840,59 +897,162 @@ namespace vcpkg::PostBuildLint
-                Checks::msg_exit_with_message(VCPKG_LINE_INFO,
-                                              msg::format(msgCommandFailed, msg::command_line = cmd_line.command_line())
-                                                  .append_raw('\n')
-                                                  .append_raw(maybe_output.error()));
+                fail |= this_lib.has_static_release;
+                fail |= this_lib.has_dynamic_release;
+            }
+
+            switch (build_info.crt_linkage)
+            {
+                case LinkageType::DYNAMIC:
+                    fail |= this_lib.has_static_debug;
+                    fail |= this_lib.has_static_release;
+                    break;
+                case LinkageType::STATIC:
+                    fail |= this_lib.has_dynamic_debug;
+                    fail |= this_lib.has_dynamic_release;
+                    break;
+                default: Checks::unreachable(VCPKG_LINE_INFO);
+            }
+
+            if (fail)
+            {
+                libs_with_invalid_crt.push_back(std::move(this_lib));
             }
         }
 
         if (!libs_with_invalid_crt.empty())
         {
-            msg::println_warning(msgPortBugInvalidCrtLinkage, msg::expected = expected_build_type.to_string());
-
+            msg::println_warning(msgPortBugInvalidCrtLinkage,
+                                 msg::expected = format_linkage(build_info.crt_linkage, expect_release));
             for (const BuildTypeAndFile& btf : libs_with_invalid_crt)
             {
-                msg::write_unlocalized_text_to_stdout(
-                    Color::warning, fmt::format("    {}: {}\n", btf.file, btf.build_type.to_string()));
+                msg::print(
+                    LocalizedString().append_indent().append(msgPortBugInvalidCrtLinkageEntry, msg::path = btf.file));
+                LocalizedString prefix;
+                if ((btf.has_dynamic_debug + btf.has_dynamic_release + btf.has_static_debug + btf.has_static_release) ==
+                    1)
+                {
+                    prefix.append_raw(" ");
+                }
+                else
+                {
+                    msg::println();
+                    prefix.append_indent(2);
+                }
+
+                if (btf.has_dynamic_debug)
+                {
+                    msg::println(LocalizedString(prefix).append(msgLinkageDynamicDebug));
+                }
+
+                if (btf.has_dynamic_release)
+                {
+                    msg::println(LocalizedString(prefix).append(msgLinkageDynamicRelease));
+                }
+
+                if (btf.has_static_debug)
+                {
+                    msg::println(LocalizedString(prefix).append(msgLinkageStaticDebug));
+                }
+
+                if (btf.has_static_release)
+                {
+                    msg::println(LocalizedString(prefix).append(msgLinkageStaticRelease));
+                }
             }
```

Comment:
This might be reasonable to extract as an algorithm:
```cpp
/// 0 items - Does nothing
/// 1 item - .append_raw(" ").append(item)
/// 2+ items - foreach: .append_raw("\n").append_indent(indent).append(item)
LocalizedString& LocalizedString::append_floating_list(int indent, View<LocalizedString> items);
```

Reply:
Done!